packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -169,71 +169,61 @@ function handleNodeJSGlobals(
 	build: PluginBuild,
 	inject: Record<string, string | string[]>
 ) {
-	const UNENV_GLOBALS_RE = /_virtual_unenv_global_polyfill-([^.]+)\.js$/;
+	const UNENV_GLOBALS_RE = /_virtual_unenv_global_polyfill-(.+)$/;
 	const prefix = nodePath.resolve(
 		getBasePath(),
 		"_virtual_unenv_global_polyfill-"
 	);
 
+	/**
+	 * Map of module identifiers to
+	 * - `injectedName`: the name injected on `globalThis`
+	 * - `exportName`: the export name from the module
+	 * - `importName`: the imported name
+	 */
+	const injectsByModule = new Map<
+		string,
+		{ injectedName: string; exportName: string; importName: string }[]
+	>();
+
+	const moduleIdSpecifier = new Map<string, string>();
```

Comment:
It was not clear to me what this map is for. Perhaps renaming would help?

```suggestion
	const virtualModuleToModuleSpecifierMap = new Map<string, string>();
```

I think we could simplify things a bit if the keys of this map contained the whole path (i.e. `${prefix}${id}`), which would mean we don't need to use the regex below to extract that bit from the path, we can just use it directly.